text/0000-multi-part-examples-in-rustdoc/index.md

```diff
@@ -0,0 +1,231 @@
+
+- Interaction with doc-test related tags seems to be fully explained in the previous section.
+- Interaction with langauge tags seems to be explained, other than the specific error details.
+
+> - It is reasonably clear how the feature would be implemented.
+
+We are not familiar with the code base.
+
+TODO:
+- UI indication mockup
+- Doc-tests details such as line numbers and test names
+- Ordering / nesting / interleaving errors details
+- Inconsistent langauge error details
+
+We would like some more feedback before continuing with greater detail.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## New code and possible complexity
+
+This feature would introduce some amount of new code. We are not familiar with the code base to determine whether the complexity of the architecture must be increased.
+
+## Using non-standard markdown
+
+This feature adds some new rust-specific tags to markdown code blocks. Markdown itself would not be aware of this feature. Therefore, markdown tools would not be aware of it. For example, after a long code example is broken into multiple parts, an IDE will not see a complete code example, unless it supports this feature. If the documentation author cares about this, then they don't have to use this feature.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Why not an implicit `split-continue`?
+
+We have considered the possibility of the `split-continue` tag either being implicit and therefore possibly omitted or not having such a tag at all.
+Every code block between a `split-start` and a `split-end` would be considered another part of the multi-part example.
+The problem with this design is that without further syntax, it disallows having unrelated code blocks in between multi-part example code blocks.
+For example, if between the `split-start` and `split-end` of a multi-part example, the author wishes to insert a `text` code block—that shows input/output or example data—this would either not be permitted or it will be made possible with some additional logic.
+Both _not permitting_ such nested single-part examples and _additional logic_ for permitting them seems less elegant/desirable than the simpler approach of requiring explicit `split-continue` tags.
+
+## Why error on inconsistent language tags
+
+From the guide section earlier:
+
+> Language is not automatically inherited from the one in the `split-start` block. If any part has a language tag that is different than the one in the `split-start`, the documentation will fail to build.
+
+An alternative to this is that specifying the language in the `split-start` block suffices and rustdoc can understand that all following parts of the multi-part example are of the same language.
+There is a problem with that.
+Pure markdown doesn't know and will never know anything about this feature.
+Markdown only [acknowledges](commonmark-fenced) the _typical_ use of the first word to specify a language:
```

Comment:
I don't think this is a giant drawback - Rustdoc already defaults the language to `rust`, which no other markdown parser does. I would rather be more user friendly than try to go for 100% compatibility, especially since rustdoc supports markdown extensions (like intra-doc links) no other markdown parser does.

Reply:
Resolved in favor of https://github.com/rust-lang/rfcs/pull/3081/files#r578305574.